test/jdk/java/foreign/NativeTestHelper.java

```diff
@@ -123,6 +123,25 @@ public static boolean isPointer(MemoryLayout layout) {
      */
     public static final AddressLayout C_POINTER = ValueLayout.ADDRESS
             .withTargetLayout(MemoryLayout.sequenceLayout(Long.MAX_VALUE, C_CHAR));
+    /**
+     * The layout for the {@code size_t} C type
+     */
+    public static final ValueLayout C_SIZE_T = ValueLayout.ADDRESS.byteSize() == 8 ? C_LONG_LONG : C_INT;
```

Comment:
Can't you set SIZE_T using the canonical layout API?

Reply:
True. We could change the other layouts in this file over to that too.

Reply:
I've update this list of layouts to use canonicalLayouts() instead